jerry-core/ecma/base/ecma-helpers-string.c

```diff
@@ -176,17 +176,31 @@ ecma_new_ecma_string_from_utf8 (const lit_utf8_byte_t *string_p, /**< utf-8 stri
   JERRY_ASSERT (string_size > 0);
 
   ecma_string_t *string_desc_p = ecma_alloc_string ();
-  string_desc_p->refs_and_container = ECMA_STRING_CONTAINER_HEAP_CHUNKS | ECMA_STRING_REF_ONE;
   string_desc_p->hash = lit_utf8_string_calc_hash (string_p, string_size);
 
   string_desc_p->u.common_field = 0;
 
-  const size_t data_size = string_size + sizeof (ecma_string_heap_header_t);
-  ecma_string_heap_header_t *data_p = (ecma_string_heap_header_t *) mem_heap_alloc_block (data_size);
-  data_p->size = (uint16_t) string_size;
-  data_p->length = (uint16_t) lit_utf8_string_length (string_p, string_size);
-  memcpy (data_p + 1, string_p, string_size);
-  ECMA_SET_NON_NULL_POINTER (string_desc_p->u.collection_cp, data_p);
+  ecma_length_t string_length = lit_utf8_string_length (string_p, string_size);
```

Comment:
Just a question: what is ensuring that string_size is <= UINT16_MAX? Somewhere probably there is a check in the code, I am just curious where. Probably extending the JERRY_ASSERT (string_size > 0); asserty above wouldn't hurt.